packages/graphql-client/src/graphqlClient.ts

```diff
@@ -0,0 +1,143 @@
+import { Client, createClient, fetchExchange, gql } from "@urql/core";
+import {
+  BlockHeader,
+  BoxQuery,
+  BoxWhere,
+  ChainClientBox,
+  IChainDataClient,
+  SignedTransaction,
+  UnsignedTransaction
+} from "packages/common/src";
+import { HexString } from "sigmastate-js/main";
+
+export type ExtendedBoxWhere = BoxWhere & {
+  /** Base16-encoded BoxIds */
+  boxIds?: HexString[];
+
+  /** Base16-encoded ErgoTrees or Base58-encoded addresses */
+  contracts?: HexString[];
+};
+
+export type ExtendedBoxQuery = BoxQuery<ExtendedBoxWhere>;
+
+export class GraphqlClient implements IChainDataClient<BoxWhere> {
+  private _client: Client;
+
+  constructor() {
+    this._client = createClient({
+      url: "TODO",
+      exchanges: [fetchExchange],
+      requestPolicy: "network-only"
+    });
+  }
+  async getUnspentBoxes(q: BoxQuery<ExtendedBoxWhere>): Promise<ChainClientBox[]> {
+    const query = gql<{ boxes: Omit<ChainClientBox, "confirmed">[] }, QueryBoxesArgs>`
+      query boxes(
+        $boxId: String
+        $boxIds: [String!]
+        $contract: String
+        $contracts: [String!]
+        $template: String
+        $tokenId: String
+        $skip: Int
+        $take: Int
+      ) {
+        boxes(
+          boxId: $boxId
+          boxIds: $boxIds
+          ergoTree: $contract
+          ergoTrees: $contracts
+          ergoTreeTemplateHash: $template
+          tokenId: $tokenId
+          skip: $skip
+          take: $take
+        ) {
+          boxId
+          transactionId
+          index
+          value
+          creationHeight
+          ergoTree
+          assets {
+            tokenId
+            amount
+          }
+          additionalRegisters
+        }
+      }
+    `;
+
+    const response = await this._client.query(query, {
+      boxId: q.where.boxId,
+      boxIds: q.where.boxIds,
+      address: q.where.contract,
+      addresses: q.where.contracts,
+      ergoTreeTemplateHash: q.where.template,
+      tokenId: q.where.tokenId,
+      skip: q.skip,
+      take: q.take,
+      spent: false
+    });
+
+    // What to do about confirmed?
+    return response.data?.boxes.map((box) => ({ ...box, confirmed: true })) ?? [];
+  }
+  async getLastHeaders(count: number): Promise<BlockHeader[]> {
+    const query = gql<{ blockHeaders: Header[] }, { count: number }>`
+      query blockHeaders($count: Int) {
+        blockHeaders(take: $count) {
+          headerId
+          timestamp
+          version
+          adProofsRoot
+          stateRoot
+          transactionsRoot
+          nBits
+          extensionHash
+          powSolutions
+          height
+          difficulty
+          parentId
+          votes
+          extension {
+            headerId
+          }
+          adProof {
+            headerId
+          }
+        }
+      }
+    `;
+
+    const response = await this._client.query(query, { count });
+
+    // What is size? Is it the block size?
+    // What is `transactionsId`?
```

Comment:
These are non-required properties and can be safely ignored, but `BlockHeader` is mistakenly defining it as required, can you please put a `?` in `size`, `extensionId`, `transactionsId` and `adProofsId` in `BlockHeader` type, so you can ignore them from here?

Reply:
Okay, Done!